teamcity/datadog_checks/teamcity/teamcity_rest.py

```diff
@@ -0,0 +1,338 @@
+        'headers': {'name': 'headers', 'default': {"Accept": "application/json"}},
+    }
+
+    def __init__(self, name, init_config, instances):
+        super(TeamCityRest, self).__init__(name, init_config, instances)
+        self.current_build_config = self.instance.get('build_configuration', None)
+        self.instance_name = self.instance.get('name', '')
+        self.host = self.instance.get('host_affected') or self.hostname
+        self.is_deployment = is_affirmative(self.instance.get('is_deployment', False))
+        self.basic_http_auth = is_affirmative(
+            self.instance.get('basic_http_authentication', bool(self.instance.get('password', False)))
+        )
+
+        self.monitored_projects = self.instance.get('projects', {})
+        self.default_build_configs_limit = self.instance.get('default_build_configs_limit', DEFAULT_BUILD_CONFIGS_LIMIT)
+        self.default_projects_limit = self.instance.get('default_projects_limit', DEFAULT_PROJECTS_LIMIT)
+        self.global_build_configs_include = self.instance.get('global_build_configs_include', [])
+        self.global_build_configs_exclude = self.instance.get('global_build_configs_exclude', [])
+        self.collect_events = is_affirmative(self.instance.get('collect_events', True))
+        self.collect_build_metrics = is_affirmative(self.instance.get('build_config_metrics', True))
+        self.collect_test_health_checks = is_affirmative(self.instance.get('tests_health_check', True))
+        self.collect_build_problem_health_checks = is_affirmative(self.instance.get('build_problem_health_check', True))
+        self.projects_refresh_interval = self.instance.get(
+            'projects_refresh_interval', DEFAULT_PROJECTS_REFRESH_INTERVAL
+        )
+        self.projects_last_refresh = None
+
+        self.auth_type = 'httpAuth' if self.basic_http_auth else 'guestAuth'
+        self.tags = set(self.instance.get('tags', []))
+        self.build_tags = []
+
+        server = self.instance.get('server')
+        self.server_url = normalize_server_url(server)
+        self.base_url = "{}/{}".format(self.server_url, self.auth_type)
+
+        instance_tags = [
+            'server:{}'.format(sanitize_server_url(self.server_url)),
+        ]
+        if self.instance_name:
+            instance_tags.extend(['instance_name:{}'.format(self.instance_name)])
+        self.tags.update(instance_tags)
+
+        self.bc_store = BuildConfigs()
+
+        if PY2:
+            self.check_initializations.append(self._validate_config)
+
+    def _validate_config(self):
+        if self.instance.get('build_configuration') and self.instance.get('projects'):
+            raise ConfigurationError('Only one of `projects` or `build_configuration` may be configured, not both.')
+        if self.instance.get('build_configuration') is None and self.instance.get('projects') is None:
+            raise ConfigurationError('`projects` must be configured.')
```

Comment:
This the error here read `either projects or build_configuration must be configured`?

Reply:
I ended up removing this for PY2, but yes I can update that for the PY3 config validator